ExpressionEngine 7.2.0

intoeetive opened the pull request.

intoeetive committed:
display revision info on entry edit

EECORE-1760

intoeetive committed:
display version number when viewing revision

EECORE-1760

intoeetive committed:
simplify counting to current version number

intoeetive committed:
add alert banner indicating you're viewing a revision

EECORE-1760

CreepCaster committed:
backup database views

CreepCaster committed:
backup database views

eric-lamb62 committed:
adds base Add-on Controller layer

eric-lamb62 committed:
allows for specifying of the Controller namespace and defaulting to Add-on config

eric-lamb62 committed:
updates namespace pathing on extensions to not be bad

eric-lamb62 committed:
updates mcp routing to take the new Mcp Controller into account
- still keeps priority on exsting MCP methods

eric-lamb62 committed:
adds magic for add-on name param on Mcp routes

eric-lamb62 committed:
removes `Routes` as namespace magic

eric-lamb62 committed:
general tidying up

eric-lamb62 committed:
cleans up docblocks

eric-lamb62 committed:
changes injection of add-on name to route dispatch

eric-lamb62 committed:
adds getBaseUrl method to compile the URL into the proper one

eric-lamb62 committed:
adds copyright docblock and adds some minor tests

eric-lamb62 committed:
throws exception when addon_name property isn't set when namespace compilation happens

eric-lamb62 committed:
reverts phpunit.xml back to what it should be

eric-lamb62 committed:
removes check for route namespace within object build (is handled within the method)

eric-lamb62 committed:
adds more tests

eric-lamb62 committed:
removes typing

intoeetive committed:
update CKEditor to v35

added CodeBlock button
added ImageCaption button
reorganized image alignment buttons

eric-lamb62 committed:
removes add-on special case and typing

intoeetive committed:
support using external stylesheet in RTE using CKEditor

intoeetive committed:
support using external stylesheet in RTE using Redactor

intoeetive committed:
update Redactor to v3.5.2

intoeetive committed:
fix issues with Redactor build

intoeetive committed:
add inputs for RTE advanced configuration

Yulyaswan committed:
added space between textarea text and list number

intoeetive committed:
Revert "added space between textarea text and list number"

This reverts commit c17eae2ec4c02a99d7ae9713a0f448a08e624f32.

intoeetive committed:
save RTE config JSON

dynamic CodeMirror initialization

intoeetive committed:
advanced configuration for CkEditor RTE

CreepCaster committed:
backup database views

intoeetive committed:
support RTL text direction in Rich Text fields

intoeetive committed:
enable loading plugin javascript for RTE

Channel Form styling fixes for Redactor

intoeetive committed:
add CKEditor source to the main repo

intoeetive committed:
clear RTE Pages cache when Structure page is modified

intoeetive committed:
fix loading Redactor settings

intoeetive committed:
styling updates

intoeetive committed:
fix the build script

intoeetive committed:
Added "Upload Directory" column for File Manager table; #98

intoeetive committed:
Enabled entering HTML line break tags into Entry Title fiels; #1633

intoeetive committed:
If the field belongs to group, redirect to group fields list after saving

intoeetive committed:
Fixed the issue where empty edit buttons were present for sidebar elements that member does not have access to edit

intoeetive committed:
Fixed the issue where File link was not in navigation sidebar for some members with proper permissions

intoeetive committed:
Added "ungrouped" as a filter to the list of Fields; #516

intoeetive committed:
Show in which channels field is used on field edit page; #919

EECORE-2120

intoeetive committed:
fix getting correct mtime for some js files in combo loader

intoeetive committed:
Added extra confirmation toggle when deleting fieldtype that is used by fields; #484

EECORE-2128

Yulyaswan committed:
added edit entry button to relationship items

Yulyaswan committed:
Fixed Field_group test

intoeetive committed:
enable parsing linebreaks for title on frontent

intoeetive committed:
PATH_JAVASCRIPT_BUILD should always point to build, not src

Yulyaswan committed:
Removed console.log

Yulyaswan committed:
fixed issue when using filepicker for textarea. Remove possibility to upload new files in modal window

Yulyaswan committed:
fixed reloading page if entry editing from modal window

Yulyaswan committed:
Fixed filepicker modal upload functionality for textarea and RTE

Yulyaswan committed:
Limited clickable area to input or text for checkboxes and radio buttons

Yulyaswan committed:
fixed tests

Yulyaswan committed:
Prevent hiding Redactor toolbar behind entry tabs when sticked to the top of the screen

intoeetive committed:
Resolved #2410 where some fieldtypes could not be used as columns in Entry Manager

Yulyaswan committed:
moved selected checkboxes to the top of the list if list longer than tooMany value

matthewjohns0n committed:
Namespaced the Structure core nav parser to avoid conflicts with the EEHarbor codebase

intoeetive committed:
Resolved #2340 where editing template by member without settings & access permission was resetting allowed roles

EECORE-2145

Yulyaswan committed:
added dataType json to new function

intoeetive committed:
Resolved #2431 where Image HTML button should not have been editable

EECORE-2144

Yulyaswan committed:
fixed code to display image in correct place for textarea

Yulyaswan committed:
Show edit button only if reationship isn't in a modal window

intoeetive committed:
Resolved #2438 when Low Variable types were not migrated properly to Pro Variable types

intoeetive committed:
ensuse modal context is passed to Relationship field, even in Fluid

intoeetive committed:
run tests using Chrome beta (#2445)

* run tests using Chrome beta

* chrome:beta is not available, use canary

* install Canary for tests

* canary not supported by platform, run beta

* run without beta flag

* some tests are still not updated properly, should remain skipped

intoeetive committed:
Merge pull request #2339 from ExpressionEngine/feature/7.x/rte-updates

RTE updates

intoeetive committed:
Merge pull request #2345 from ExpressionEngine/feature/7.x/br-wbr-in-entry-title

Enabled entering HTML line break tags into Entry Title fiels; #1633

intoeetive committed:
Merge pull request #2358 from ExpressionEngine/feature/display-revision-info-on-entry-edit

Added visual indication when viewing entry version; #1897

intoeetive committed:
Merge pull request #2419 from ExpressionEngine/bug/7.x/large-checkbox-area-EECORE-2140

Resolved #2403 where clickable area for checkboxes was too large

intoeetive committed:
Merge pull request #2378 from ExpressionEngine/feature/7.x/extra-delete-confirmation-for-fieldtypes-that-have-fields

Added extra confirmation toggle when deleting fieldtype that is used by fields; #484

intoeetive committed:
Merge branch '7.dev' into feature/7.x/edit-relationship-page-from-entry-EECORE-2116

# Conflicts:
#	themes/ee/asset/javascript/src/fields/rte/redactor/redactor.min.js
#	themes/ee/asset/javascript/src/fields/rte/redactor/redactor.min.js.map

intoeetive committed:
added EE 6.4.0 update file

intoeetive committed:
update controllers list for test

Yulyaswan committed:
fixed test

matthewjohns0n committed:
Disallow phar uploads

intoeetive committed:
add Pro fronteditor source into core

intoeetive committed:
fixed php 8.1 compatibility issue with pro search

intoeetive committed:
Resolved #2435 when having two Pro Search results tags with `keywords:stem` parameter on same page caused PHP error

intoeetive committed:
fix PHPUnit test for add-ons controller

intoeetive committed:
fix PHPUnit test for add-ons controller

(cherry picked from commit 547528d396d0f0100cd6bcdb1d681af2075fdab0)

Yulyaswan committed:
resolved js conflict

intoeetive committed:
Merge pull request #2421 from ExpressionEngine/bug/7.x/redactor-toolbar-issue-EECORE-2141

Resolved #2361 where Redactor toolbar was hidden behind entry tabs when sticked to the top of the screen

intoeetive committed:
Update the license mentions of Pro components

intoeetive committed:
Merge pull request #2380 from ExpressionEngine/feature/7.x/edit-relationship-page-from-entry-EECORE-2116

Added ability to edit related entries content from relationship field; #656

matthewjohns0n committed:
Merge pull request #2456 from ExpressionEngine/feature/7.x/license-update

Update the license mentions of Pro components

matthewjohns0n committed:
Merge pull request #2444 from ExpressionEngine/bug/7.x/low-vars-to-pro-vars-type-update

Resolved #2438 when Low Variable types were not migrated properly to Pro Variable types

matthewjohns0n committed:
Merge branch '7.dev' into feature/6.x/object-addons

intoeetive committed:
bring 6.4 update file up-to-date

intoeetive committed:
remove frontedit src from build

intoeetive committed:
Merge pull request #2454 from ExpressionEngine/bug/7.x/php-8.1-pro-search-error

fixed php 8.1 compatibility issue with pro search

intoeetive committed:
Merge pull request #2448 from CreepCaster/bug/backup-for-database-views

Enable including database views into backups

bryannielsen committed:
Merge pull request #2413 from ExpressionEngine/bug/7.x/f-manager-modal-issue-EECORE-2138

Resolved #2411, #2412 where filepicker was not always working outside of File field

matthewjohns0n committed:
Merge pull request #2458 from ExpressionEngine/feature/7.x/update-6.4-updater

bring 6.4 update file up-to-date

Yulyaswan committed:
fixed grid view for resize page

matthewjohns0n committed:
Merge pull request #2367 from mithra62/feature/6.x/object-addons

Feature/6.x/object addons

matthewjohns0n committed:
Merge pull request #2453 from ExpressionEngine/feature/7.x/pro-src-into-core

add Pro fronteditor source into core

intoeetive committed:
do not show assigned channels for new field

bryannielsen committed:
Resolved #2431 image HTML button not editable in profile

matthewjohns0n committed:
Merge pull request #2471 from ExpressionEngine/bug/7.x/profile-image-html-button-editable

Resolved #2431 image HTML button not editable in profile

robinsowell commented:
I get a javascript error on the publish page if I have a textarea field with the default html buttons on it.  Only field in my test channel, I start typing in the text area, hit enter, get
<img width="1566" alt="Screen Shot 2022-10-26 at 1 29 58 PM" src="https://user-images.githubusercontent.com/1181219/198112157-be3f530d-782b-44c1-9431-ee9a46d38bf1.png">

Andy confirmed it.  I tested in 7.1 and same exact test didn't trigger the error.

bryannielsen committed:
Resolved #2263 fix block allow instructions

Yulyaswan committed:
Аixed display of checkboxes on the Channel Status

Yulyaswan committed:
fixed JS issue for textarea if html buttons are on

bryannielsen committed:
Merge pull request #2475 from ExpressionEngine/bug/7.x/channel-status-checkbox-view-issue

Fixed display of checkboxes on the Channel page Status tab

bryannielsen committed:
Merge pull request #2477 from ExpressionEngine/bug/7.x/textarea-js-issue

Fixed JS issue for textarea if html buttons are on

matthewjohns0n committed:
Merge pull request #2473 from ExpressionEngine/bug/7.x/block-allow-instruction-fix

Resolved #2263 fix block allow instructions

tknaller committed:
fix recaptcha

robinsowell commented:
I'm getting an error trying to use the RTE on the frontend.

Initially had the field set to Redactor.  If I have a custom JS template set, the combo load returns an error "No such property: 'JS_TMPL' on ExpressionEngine\Legacy\Facade"

If I switch the field to use a different Redactor toolbar, I don't get that error, but I get a js error.
<img width="1068" alt="Screen Shot 2022-10-28 at 5 20 26 PM" src="https://user-images.githubusercontent.com/1181219/198734801-3ceb8341-88c3-48fa-82af-4efb0ea96ae1.png">

intoeetive committed:
fix loading custom JS for RTE on frontend

intoeetive committed:
Resolved #2484 where FTP library was not working properly on PHP 8.1

Yulyaswan committed:
fixed channel form issue when used RTE

robinsowell committed:
Added default empty string to placeholder variable

php 8.1, eeh rating module was throwing:

```Deprecated
htmlentities(): Passing null to parameter #1 ($string) of type string is deprecated

ee/ExpressionEngine/View/_shared/filters/filter.php, line 17```

Basically, the collection filter didn't have a default specified, so it was null and htmlentities didn't love it.

Seemed better to put a default here than require it of add-ons.

matthewjohns0n committed:
Update Generate add-on to use EEObjects and maintain a compatibility mode

intoeetive committed:
extra `isset` checks to enable early parsing of javascript in templates for RTE

intoeetive committed:
do not include webpack config into build

intoeetive committed:
Resolved #2412 where using File Picker without File field was causing JavaScript error

bryannielsen committed:
Merge pull request #2493 from ExpressionEngine/feature/7.x/filepicker-no-file-field

Resolved #2412 where using File Picker without File field was causing JavaScript error

bryannielsen committed:
Merge pull request #2492 from ExpressionEngine/bug/7.x/rte-js-early-parsing

extra `isset` checks to enable early parsing of javascript in templates for RTE

bryannielsen committed:
Update success banner for 7.2

matthewjohns0n committed:
Add make:tag and make:action cli commands

matthewjohns0n committed:
Create make:extension-hook cli command

matthewjohns0n committed:
Update to use the correct hook name

bryannielsen committed:
Merge pull request #2494 from ExpressionEngine/feature/7.dev/update-cli-commands

Update Generate add-on to use EEObjects and maintain a compatibility mode
